web/src/components/repo/settings/SecretsTab.vue

```diff
@@ -69,10 +69,33 @@ async function loadSecrets(page: number): Promise<Secret[] | null> {
     throw new Error("Unexpected: Can't load repo");
   }
 
-  return apiClient.getSecretList(repo.value.id, page);
+  const globalSecrets = (await apiClient.getGlobalSecretList(page)) ?? [];
+  const orgSecrets = (await apiClient.getOrgSecretList(repo.value.org_id, page)) ?? [];
+  const repoSecrets = (await apiClient.getSecretList(repo.value.id, page)) ?? [];
+  return [...globalSecrets, ...orgSecrets, ...repoSecrets];
 }
 
-const { resetPage, data: secrets } = usePagination(loadSecrets, () => !selectedSecret.value);
+const { resetPage, data: _secrets } = usePagination(loadSecrets, () => !selectedSecret.value);
+const secrets = computed(() => {
+  const secretsList: Record<string, Secret & { edit?: boolean }> = {};
+
+  // eslint-disable-next-line no-restricted-syntax
+  for (const level of ['repo', 'org', 'global']) {
```

Comment:
this should be handled in the backend as you could have 5 pages of secrets in global and only one in repo that overrides global secret that is in last page or you can have other way around... there is way too many things that can go wrong here

Reply:
But as written before isn't it the same for the backend. As long as we use pagination we would run into this or we would have to query additionally for each level if the above level has the same names.

Reply:
not really, why would 2nd step be needed, if showing repo secrets we don't need secrets with same name as they would be overriden and won't be shown anyway

Reply:
Okay adjusted the code to do this now.